### PR TITLE
Header

### PR DESCRIPTION
**Pull request by navid9696**
Closes: #18

### CODE REVIEW

File: src/components/AccountSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			case 'email':
+				return <EmailForm setIsDetailsVisible={setIsFormVisible} />
+			case 'password':
+				return <PasswordForm setIsDetailsVisible={setIsFormVisible} />
+			case 'address':
+				return <AddressForm setIsDetailsVisible={setIsFormVisible} />
+			default:
+				return null
+		}
+	}
```
Comment (bart-krakowski) [inline src/components/AccountSettings.tsx:60]: Pls move it outside the component and add a param to specify the `contentId`

File: src/components/CategoriesTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Tabs } from '@mui/material'
+import CategoryBtn from './CategoryBtn'
+import { categoriesData } from '~/data/categories'
+import { useState } from 'react'
+
+const CategoriesTabs = () => {
+	const [activeCategory, setActiveCategory] = useState<null | number>(0)
+	const [activeSubCategory, setActiveSubCategory] = useState<null | number>(null)
```
Comment (bart-krakowski) [inline src/components/CategoriesTabs.tsx:8]: Those state units are coupled, so consider using a reducer

**Review by bart-krakowski — CHANGES_REQUESTED** [review]

File: src/components/CategoriesTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	activeCategory: number
+	activeSubCategory: number | null
+}
+
+const initialState = {
+	activeCategory: 0,
+	activeSubCategory: null,
+}
+
+const reducer = (state: State, action: Action): State => {
```
Comment (bart-krakowski) [inline src/components/CategoriesTabs.tsx:25]: Protip: you can use this package https://github.com/piotrwitek/typesafe-actions
Not-blocking

File: src/components/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		accessibility: false,
+		adaptiveHeight: true,
+		arrows: false,
+		autoplaySpeed: 5000,
+		infinite: true,
+		speed: 1500,
+		slidesToShow: 1,
+		slidesToScroll: 1,
+		autoplay: true,
+	}
```
Comment (bart-krakowski) [inline src/components/Header.tsx:17]: Pls move it outside

File: src/components/NicknameForm.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		.min(5, { message: 'Make sure your name is at least 5 characters long.' })
+		.max(15, { message: 'Keep your name within 15 characters.' }),
+})
+
+interface NicknameFormInput {
+	nickname: string
+}
+
+const NicknameForm = ({ setIsDetailsVisible }: SettingFormsProps) => {
+	const [isFocusedField, setIsfocusedField] = useState(false)
```
Comment (bart-krakowski) [inline src/components/NicknameForm.tsx:25]: Out of curiosity, can you use `setFocus` from the useForm hook?
https://react-hook-form.com/docs/useform/setfocus

**Review by bart-krakowski — CHANGES_REQUESTED** [review]

File: src/components/NicknameForm.tsx (hunk 5 above)
Comment (navid9696) [reply src/components/NicknameForm.tsx]: I tried to use `setFocus`, but to unfocus I would have to use `ref` probably, and I still needed the focus state to dynamically render the `InputAdornment`

File: src/components/CategoriesTabs.tsx [hunk 6]
```diff
@@ -1,42 +1,45 @@
...
+		case getType(setActiveSubCategory):
 			return {
 				...state,
-				activeSubCategory: payload,
+				activeSubCategory: action.payload.activeSubCategory,
 			}
-
 		default:
 			return state
 	}
```
Comment (bart-krakowski) [inline src/components/CategoriesTabs.tsx]: You can simplify this code by using this syntax.
https://github.com/piotrwitek/typesafe-actions?tab=readme-ov-file#using-createreducer-api-with-type-free-syntax

**Review by bart-krakowski — CHANGES_REQUESTED** [review]

**Review by bart-krakowski — APPROVED** [review]

**Pull request merged**